File: app/src/main/java/com/example/biblio/DarkThemeApplication.java
package com.example.biblio;

import android.app.Application;
import android.content.SharedPreferences;

import androidx.preference.PreferenceManager;

import com.example.biblio.helpers.ThemeHelper;

public class DarkThemeApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();

        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
        boolean themePref = preferences.getBoolean(getResources().getString(R.string.theme_pref), false);

        ThemeHelper.applyTheme(themePref);
    }
}

File: app/src/main/java/com/example/biblio/DarkThemeApplication.kt
package com.example.biblio

import android.app.Application
import androidx.preference.PreferenceManager
import com.example.biblio.helpers.ThemeHelper

class DarkThemeApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        val themePref = PreferenceManager.getDefaultSharedPreferences(this)
                .getBoolean(resources.getString(R.string.theme_pref), false)
        ThemeHelper.applyTheme(themePref)
    }
}